python/mleap/sklearn/preprocessing/data.py

```diff
@@ -624,18 +629,29 @@ def __init__(self):
         super(OneHotEncoderSerializer, self).__init__()
 
     def serialize_to_bundle(self, transformer, path, model_name):
+        if transformer.drop_last and transformer.handle_unknown == 'ignore':
+            raise NotImplementedError("MLeap's OneHotEncoder cannot drop the last feature column while also ignoring unknown features")
+        if len(transformer.categories_) != 1:
+            raise NotImplementedError("MLeap can only one-hot encode a single column at a time")
+        singular_categories = transformer.categories_[0]
+        if not np.array_equal(singular_categories, np.arange(singular_categories.size)):
+            raise ValueError(f"Categories {singular_categories} do not form a valid index range")
+        if transformer.drop is not None:
+            raise NotImplementedError("Scikit-learn's OneHotEncoder `drop` parameter is not supported by MLeap")
+        if transformer.dtype != np.float64:
+            raise NotImplementedError("Scikit-learn's OneHotEncoder `dtype` parameter is not supported by MLeap")
 
-        # compile tuples of model attributes to serialize
         attributes = list()
-        attributes.append(('size', transformer.n_values_.tolist()[0]))
-        # setting drop_last to True, so to that 1HE maintains parity with MLeap/Spark
-        attributes.append(('drop_last', True))
+        attributes.append(('size', singular_categories.size))
         if transformer.handle_unknown == 'ignore':
+            # MLeap's OneHotEncoderModel adds an extra column when keeping invalid data
+            # so we drop that extra column to match sklearn's ignore behavior
             attributes.append(('handle_invalid', 'keep'))
+            attributes.append(('drop_last', True))
         else:
             attributes.append(('handle_invalid', 'error'))
+            attributes.append(('drop_last', False))
```

Comment:
@ancasarb When going through this code again, I realized that we can't always do `attributes.append(('drop_last', True))` like in your PR. If we do, then `transformer.handle_unknown == 'error'` will cause MLeap to drop the last feature column, but scikit will still have that column

Reply:
👍 thanks for looking into this again, it's been quite a bit of a pain to get this right.